### ISSUE TITLE
Compile error on rust-starter template

### ISSUE THREAD

**Issue by orkunkilic** [risc0/risc0#763]
Hi, I was trying to test zkvm today but got error on compilation.

I created the project using `cargo risczero new my_project` script, so set-up should be ok. However, when I run `cargo run` I face several errors.

First:

```
method_name: error: could not compile 'cc' due to 286 previous errors
error: failed to run custom build command for 'methods v0.1.0 (/Users/x/vm_benchmark/vm_benchmark/methods)'
```

Next line:

```
Caused by:
  process didn't exit successfully: '/Users/x/projects/vm_benchmark/vm_benchmark/target/release/build/methods-e83407e47de0716f/build-script-build' (exit status: 101)
  --- stdout
  Building guest package methods.method_name
  cargo:rerun-if-env-changed=RISC0_SKIP_BUILD
  Building guest package: /Users/x/.rustup/toolchains/nightly-2023-03-06-aarch64-apple-darwin/bin/cargo build --release --target /Users/x/vm_benchmark/vm_benchmark/target/release/build/methods-11740a4c18f9cf4f/out/riscv32im-risc0-zkvm-elf.json -Z build-std=alloc,core,proc_macro,panic_abort,std -Z build-std-features=compiler-builtins-mem --manifest-path /Users/x/vm_benchmark/vm_benchmark/methods/guest/Cargo.toml --target-dir /Users/x/vm_benchmark/vm_benchmark/target/riscv-guest
  Using rust standard library root: /Users/x/vm_benchmark/vm_benchmark/target/release/build/methods-11740a4c18f9cf4f/out/rust-std_ad062567ca92a9b48981bf6d4ac3375b88abc750249689321f6ab3bbc996d11d
```

I got the same error on my M2 Mac and Intel Mac. Thanks!

**Comment by tzerrell** [comment]
This is due to https://github.com/risc0/risc0/issues/754. We have a workaround (although not yet a full fix). To use the workaround, add the `branch` and `use-git-branch` parameters to get the latest version of the rust-starter template from our repo:
```
cargo risczero new --branch main --use-git-branch main <your-project-name>
```
(The first parameter controls where you get the template from, and the latter controls what version of risc0 your project depends on -- since we've made some API changes since releasing `0.16.1` you need to update both for them to be compatible.)

If you want to use an existing project you've already created with `cargo risczero new` without using these parameters, you can instead follow the workaround instructions in #754.

I'm leaving this issue open until we have a full fix in and no workarounds are needed.

**Comment by flaub** [comment]
I think this has been fixed in 0.17, but if you still run into this issue, please feel free to file another issue.
